### PR TITLE
[Object] Restore the StrMap behavior in JSON/SHash/SEqual

### PR DESCRIPTION
**Pull request by junrushao1994**

### CODE REVIEW

File: include/tvm/node/container.h [hunk 1]
```diff
@@ -82,6 +82,21 @@ class MapNode : public Object {
   /*! \brief the data content */
   ContainerType data;
 
+  /*!
+   * \brief Check if all keys in the MapNode are instance of the given Object type
+   * \tparam T The given Object type
+   * \return The result of the check
+   */
+  template <class T>
+  bool KeysAreInstance() const {
```
Comment (tqchen) [inline include/tvm/node/container.h:91]: This is not so commonly used, let us make it an internal helper function instead, we can also use https://en.cppreference.com/w/cpp/algorithm/all_any_none_of

Comment (tqchen) [reply include/tvm/node/container.h:91]: @junrushao1994 please followup

**Comment by junrushao1994** [comment]
@tqchen Just addressed the comments and added a test case for JSON compatibility. Please take a look when you have time :-)

File: include/tvm/node/container.h (hunk 1 above)
Comment (junrushao1994) [reply include/tvm/node/container.h]: yeah I just replaced it with std::all_of. Thanks!

File: include/tvm/node/container.h [hunk 2]
```diff
@@ -96,8 +98,7 @@ class MapNode : public Object {
  * \tparam V The value NodeRef type.
  */
 template <typename K, typename V,
-          typename = typename std::enable_if<std::is_base_of<ObjectRef, K>::value ||
-                                             std::is_base_of<std::string, K>::value>::type,
+          typename = typename std::enable_if<std::is_base_of<ObjectRef, K>::value>::type,
```
Comment (tqchen) [inline include/tvm/node/container.h:101]: I think that it is better to directly change it to static_assert later,

Comment (junrushao1994) [reply include/tvm/node/container.h:101]: Yeah, but the difference is less fundamental imo

**Comment by tqchen** [comment]
Thanks @junrushao1994 !

**Pull request merged**